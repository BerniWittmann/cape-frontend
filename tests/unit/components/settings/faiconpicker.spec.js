import { mount } from '@vue/test-utils'
import { i18n } from '../../setupPlugins'

import FontAwesomeChooser from '@/components/settings/FAIconPicker.vue'
import FontAwesomeIcons from '@/utils/fontAwesomeIcons'

jest.mock('@/utils/fontAwesomeIcons', () => (
  [
    'fa-glass',
    'fa-music',
    'fa-search',
    'fa-envelope-o',
    'fa-heart'
  ]
))

describe('Components', () => {
  describe('Font Awesome Icon Picker', () => {
    let propsData
    let cmp

    beforeEach(() => {
      propsData = {
        search: ''
      }
      render()
    })

    function render() {
      cmp = mount(FontAwesomeChooser, {
        i18n,
        propsData
      })
    }

    it('renders', () => {
      expect(cmp.html()).toMatchSnapshot()
      const fontAwesomeChooserComponent = cmp.find('.font-awesome-chooser')
      expect(fontAwesomeChooserComponent.exists()).toBe(true)
    })

    it('renders the search', () => {
      expect(cmp.find('.el-input').exists()).toBe(true)
    })

    it('renders the close button', () => {
      const closeBtn = cmp.find('.el-button')
      expect(closeBtn.exists()).toBe(true)
    })

    it('renders all icons', () => {
      expect(cmp.find('icon-overview').exists())
      FontAwesomeIcons.forEach(function (icon) {
        expect(cmp.find('.' + icon).exists()).toBe(true)
      })
      expect(cmp.findAll('.icon-item').length).toEqual(FontAwesomeIcons.length)
    })

    it('renders readable icon names', () => {
      for (let i = 0; i < FontAwesomeIcons.length; i++) {
        const text = cmp.findAll('.text').at(i)
        expect(text.exists()).toBe(true)
        expect(text.text()).toEqual(FontAwesomeIcons[i].substring(3))
      }
    })

    it('can handle an empty icon list', () => {
      cmp.vm.search = 'someTextToMakeNoIconsToBeShown'
      expect(cmp.find('icon-overview').exists())
      expect(cmp.findAll('.icon-item')).toEqual({})
    })

    it('can search the icons', () => {
      cmp.vm.search = 'mus'
      expect(cmp.find('.fa-music').exists()).toBe(true)
      expect(cmp.findAll('.icon-item').length).toEqual(1)
    })

    it('emits an event if closed', () => {
      expect(cmp.emitted('close')).toBeFalsy()
      const closeBtn = cmp.find('.el-button')
      closeBtn.trigger('click')
      expect(cmp.emitted('close')).toBeTruthy()
    })

    it('emits an event with the icon if an icon is clicked', () => {
      expect(cmp.emitted('click')).toBeFalsy()
      cmp.find('.icon-item').trigger('click')
      expect(cmp.emitted('click')).toBeTruthy()
      expect(cmp.emitted('click')[0]).toEqual(['fa-glass'])
    })

    it('closes if an icon is chosen', () => {
      expect(cmp.emitted('close')).toBeFalsy()
      cmp.find('.icon-item').trigger('click')
      expect(cmp.emitted('close')).toBeTruthy()
    })
  })
})
