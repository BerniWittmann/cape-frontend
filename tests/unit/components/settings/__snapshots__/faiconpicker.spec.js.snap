// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Components Font Awesome Icon Picker renders 1`] = `
<section class="el-container font-awesome-chooser is-vertical" search="">
  <header class="el-header" style="height: 60px;">
    <div class="el-input el-input--mini el-input--prefix">
      <!----><input type="text" autocomplete="off" placeholder="Type to search" class="el-input__inner"><span class="el-input__prefix"><i class="el-input__icon el-icon-search"></i><!----></span>
      <!---->
      <!---->
    </div> <button type="button" class="el-button el-button--default el-button--mini">
      <!----><i class="el-icon-close"></i><span>Close</span></button>
  </header>
  <main class="el-main" id="icon-overview">
    <div class="icon-item">
      <div class="icon"><i id="fa-icon" class="fa fa-lg fa-glass"></i></div>
      <div class="text-space">
        <div class="text">glass</div>
      </div>
    </div>
    <div class="icon-item">
      <div class="icon"><i id="fa-icon" class="fa fa-lg fa-music"></i></div>
      <div class="text-space">
        <div class="text">music</div>
      </div>
    </div>
    <div class="icon-item">
      <div class="icon"><i id="fa-icon" class="fa fa-lg fa-search"></i></div>
      <div class="text-space">
        <div class="text">search</div>
      </div>
    </div>
    <div class="icon-item">
      <div class="icon"><i id="fa-icon" class="fa fa-lg fa-envelope-o"></i></div>
      <div class="text-space">
        <div class="text">envelope-o</div>
      </div>
    </div>
    <div class="icon-item">
      <div class="icon"><i id="fa-icon" class="fa fa-lg fa-heart"></i></div>
      <div class="text-space">
        <div class="text">heart</div>
      </div>
    </div>
  </main>
</section>
`;
